Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Api/Common/Errors/ProblemResults.cs

```diff
@@ -0,0 +1,33 @@
+﻿namespace EvolutionaryArchitecture.Fitnet.Contracts.Api.Common.Errors;
+
+using Core.Common.BussinessRules;
+using Microsoft.AspNetCore.Http;
+using static ErrorType;
+
+internal static class ProblemResults
+{
+    internal static IResult ToProblem(this IReadOnlyCollection<Error> errors)
+    {
+        var error = errors.First();
+        var statusCode = error.NumericType switch
+        {
+            BusinessRuleError.Type => StatusCodes.Status409Conflict,
+            (int)Conflict => StatusCodes.Status409Conflict,
+            (int)Validation => StatusCodes.Status400BadRequest,
+            (int)NotFound => StatusCodes.Status404NotFound,
+            (int)Failure => StatusCodes.Status500InternalServerError,
+            (int)Unexpected => StatusCodes.Status500InternalServerError,
+            (int)Unauthorized => StatusCodes.Status401Unauthorized,
+            (int)Forbidden => StatusCodes.Status403Forbidden,
+            _ => StatusCodes.Status500InternalServerError
+        };
+
+        var errorMessage = GetErrorMessage(errors);
+        var results = Results.Problem(errorMessage, statusCode: statusCode);
+
+        return results;
+    }
+
+    private const string? Separator = ", ";
+    private static string GetErrorMessage(IReadOnlyCollection<Error> errors) => string.Join(Separator, errors.Select(error => error.Description));
```

Comment:
Why not just IEnumerable? :)

Btw. Fix formatting, the line is too long

Reply:
Even though it is today only in Contracts, it would make sense to move it to our Common package. Otherwise it pollutes the current solution